docker-compose/technical/grafana/alerting/alert_rules.json

```diff
@@ -0,0 +1,137 @@
+{
```

Comment:
This is the correct format for file provisioning (if one can put this file /etc/grafana/provisioning/alerting).

If one wants to use Grafana API instead, we need to use the content of `curl 'http://localhost:7000/api/v1/provisioning/alert-rules/f154b805-ea70-4640-8ccb-71abb97e9803'`
which can then be used with a POST on our different environments `curl 'http://localhost:7000/api/v1/provisioning/alert-rules' -X POST -H 'Content-Type: application/json' -H 'X-Disable-Provenance: true' -H 'Origin: http://localhost:7000' --data-raw '{"id":2,"uid":"testuid","orgID":1,"folderUID":"bbac38da-14fd-4758-92e7-3b3d06dfc8ba","ruleGroup":"10sEval","title":"CheckReadyQueues (NEW)","condition":"C","data":[{"refId":"A","queryType":"","relativeTimeRange":{"from":21600,"to":0},"datasourceUid":"PBFA97CFB590B2093","model":{"datasource":{"type":"prometheus","uid":"PBFA97CFB590B2093"},"disableTextWrap":false,"editorMode":"builder","expr":"rabbitmq_detailed_queue_messages_ready{queue=~\".*.run.*\"}","fullMetaSearch":false,"includeNullMetadata":true,"instant":false,"interval":"","intervalMs":15000,"legendFormat":"__auto","maxDataPoints":43200,"range":true,"refId":"A","useBackend":false}},{"refId":"B","queryType":"","relativeTimeRange":{"from":21600,"to":0},"datasourceUid":"__expr__","model":{"conditions":[{"evaluator":{"params":[],"type":"gt"},"operator":{"type":"and"},"query":{"params":["B"]},"reducer":{"params":[],"type":"last"},"type":"query"}],"datasource":{"type":"__expr__","uid":"__expr__"},"expression":"A","intervalMs":1000,"maxDataPoints":43200,"reducer":"last","refId":"B","type":"reduce"}},{"refId":"C","queryType":"","relativeTimeRange":{"from":21600,"to":0},"datasourceUid":"__expr__","model":{"conditions":[{"evaluator":{"params":[5],"type":"gt"},"operator":{"type":"and"},"query":{"params":["C"]},"reducer":{"params":[],"type":"last"},"type":"query"}],"datasource":{"type":"__expr__","uid":"__expr__"},"expression":"B","intervalMs":1000,"maxDataPoints":43200,"refId":"C","type":"threshold"}}],"updated":"2024-04-11T13:00:58Z","noDataState":"NoData","execErrState":"Error","for":"30s","annotations":{"__dashboardUid__":"UplRgsDIz","__panelId__":"14","description":"The rule checks regularly if the \"run computation\" ready queues have more than X messages","summary":"If this alert is triggered, there is probably an issue with the calculation server\n- low dimensioning, the queue is now emptied fast enough\n- server crash, the queue is not emptied anymore\nIt leads to longer response time for users trying to run calculations"},"provenance":"file","isPaused":false}'`

For the moment, we don't have the necessary rights on our environments but it works locally. 

We need to think about how to manage and update these dashboards/alerts/... that we are creating, some do this with the API https://github.com/ThinkParQ/beegfs/blob/4006d2244c58889a3732aeacba8de285834ebb0a/mon/scripts/grafana/import-alerts and storing the different json as mentioned above. We could also use https://www.terraform.io/, but it's probably too heavy for our use case.

Note also that:
* with the HTTP API, if you set -H 'X-Disable-Provenance: true', the alert can be deleted, edited etc not like provisioned alerts/dashboards. 
* the folder and ruleGroup needs to exist before calling the POST.